allennlp/modules/token_embedders/bidirectional_language_model_token_embedder.py

```diff
@@ -0,0 +1,170 @@
+        num_layers = self._lm._contextualizer.num_layers + 1
+        self._scalar_mix = ScalarMix(mixture_size=num_layers, do_layer_norm=False, trainable=True)
+
+        character_dim = self._lm._text_field_embedder.get_output_dim()
+        contextual_dim = self._lm._contextualizer.get_output_dim()
+
+        if not contextual_dim % character_dim == 0 or character_dim > contextual_dim:
+            raise RuntimeError(
+                    "The output dimensions for the text_field_embedder " +
+                    f"({character_dim}) and the contextualizer ({contextual_dim})" +
+                    f" from the language model loaded from {archive_file} are " +
+                    "not compatible. Please check the config used to train that " +
+                    "model and ensure that the output dimension of the " +
+                    "text_field_embedder divides the output dimension of the " +
+                    "contextualizer.")
+        self._character_embedding_duplication_count = contextual_dim // character_dim
+
+        for param in self._lm.parameters():
+            param.requires_grad = requires_grad
+
+    def get_output_dim(self) -> int:
+        return self._lm._contextualizer.get_output_dim() # pylint: disable=protected-access
+
+    def forward(self,  # type: ignore
+                inputs: torch.Tensor) -> Dict[str, torch.Tensor]:
+        """
+        Parameters
+        ----------
+        inputs: ``torch.Tensor``
+            Shape ``(batch_size, timesteps, ...)`` of token ids representing the current batch.
+            These must have been produced using the same indexer the LM was trained on.
+
+        Returns
+        -------
+        The bidirectional language model representations for the input sequence, shape
+        ``(batch_size, timesteps, embedding_dim)``
+        """
+        # pylint: disable=arguments-differ
+        if self._bos_indices is not None:
+            mask = get_text_field_mask({"": inputs})
+            inputs, _ = add_sentence_boundary_token_ids(
+                    inputs, mask, self._bos_indices, self._eos_indices
+            )
+
+        source = {self._token_name: inputs}
+        result_dict = self._lm(source)
+
+        # shape (batch_size, timesteps, embedding_size)
+        character_embeddings = result_dict["character_embeddings"]
+        contextual_embeddings = result_dict["lm_embeddings"]
+
+        # Typically character embeddings are smaller than contextualized embeddings. Since we're
+        # averaging the character embeddings along with all the contextualized layers we need to
+        # make their dimensions match. Simply repeating the character embeddings is a crude, but
+        # effective, way to do this.
+        duplicated_character_embeddings = torch.cat(
+                [character_embeddings] * self._character_embedding_duplication_count, -1
+        )
+        contextual_embeddings.append(duplicated_character_embeddings)
```

Comment:
To be consistent with `Elmo` and the notion that `duplicated_character_embeddings` are the first layer of the model, make `duplicated_character_embeddings` the first element of the list passed into `self._scalar_mix`.